From 01115314a3c12073e5e6d4e02970867527c66e86 Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Fri, 22 Jun 2012 21:58:32 +0100
Subject: [PATCH 24/38] EPEL 5: Add 1 second pause after unmounting any filesystem.

This fixes the test-charset-fidelity test case.  A simpler example
which fails without this is:

  guestfish <<EOF
  sparse test1.img 1G
  run
  part-disk /dev/sda mbr
  mkfs msdos /dev/sda1
  mount /dev/sda1 /
  touch /foo
  umount /
  mkfs ntfs /dev/sda1
  # Next command would fail:
  mount /dev/sda1 /
  vfs-type /dev/sda1 -x
  EOF
---
 daemon/mount.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/daemon/mount.c b/daemon/mount.c
index af92834..30a8a8c 100644
--- a/daemon/mount.c
+++ b/daemon/mount.c
@@ -228,6 +228,9 @@ do_umount (const char *pathordevice,
     return -1;
   }
 
+  /* RHEL 5 only. */
+  sleep (1);
+
   return 0;
 }
 
@@ -386,6 +389,9 @@ do_umount_all (void)
 
   free_stringslen (mounts.argv, mounts.size);
 
+  /* RHEL 5 only. */
+  sleep (1);
+
   return 0;
 }
 
-- 
1.7.4.1

